test/e2e/specs/customer_admin.go

```diff
@@ -81,6 +81,50 @@ var _ = Describe("Openshift on Azure customer-admin e2e tests [CustomerAdmin][Fa
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	It("should be able to edit project-request", func() {
+		projectTemplate, err := sanity.Checker.Client.CustomerAdmin.TemplateV1.Templates("openshift").Get("project-request", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		// modify project template object
+		projectTemplate.SetAnnotations(map[string]string{
+			"foo.annotation.com": "bar",
+		})
+		_, err = sanity.Checker.Client.CustomerAdmin.TemplateV1.Templates("openshift").Update(projectTemplate)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should be able to edit self-provisioners clusterrolebinding", func() {
+		crbOriginal, err := sanity.Checker.Client.CustomerAdmin.RbacV1.ClusterRoleBindings().Get("self-provisioners", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		crb := crbOriginal.DeepCopy()
+
+		// create ClusterRoleBinding which would allow us to test edit
+		// and would not break any other tests
+		subject := []rbacv1.Subject{
+			{
+				Name: "system:authenticated:oauth",
+				Kind: "Group",
+			},
+			{
+				Name: "osa-customer-admins",
+				Kind: "Group",
+			},
+		}
+
+		crb.Subjects = subject
+
+		_, err = sanity.Checker.Client.CustomerAdmin.RbacV1.ClusterRoleBindings().Update(crb)
+		Expect(err).ToNot(HaveOccurred())
+
+		// HACK: openshift reconciler should not touch this role because of:
+		// openshift.io/reconcile-protect: "true"
+		// we need to confirm this
+		time.Sleep(5 * time.Second)
```

Comment:
Why is this sleep necessary? The comment doesn't make this any clearer to me.

Reply:
This cole by default is managed by Openshift core rbac reconciler loop. we exclude it. This gives time for the controller to react and check if "it needs to change it". Our expectation is that we gonna manage it from now on. So we just checking if it does not touch it.

Reply:
+1 with @ehashman - I thought this reconcile happened at apiserver start?  Do you see an issue without this sleep?

Reply:
I see that sometimes our process runs faster than apiserver reacts to `watch` event and patches/not patches the object. In theroy we could read the object and see - "all good, this is what we submitted in" but in next half-second reconcile runs and changes it to default back. This  is my understanding of the process how it work (controller and reconcile loop). If this is not a case - I need to update my knowledge :)